src/sagas/bonding-curve.js

```diff
@@ -0,0 +1,67 @@
+import { takeLatest, call, select } from 'redux-saga/effects'
+
+import * as bondingCurveActions from '../actions/bonding-curve'
+import * as walletSelectors from '../reducers/wallet'
+import { bondingCurve } from '../bootstrap/dapp-api'
+import { lessduxSaga } from '../utils/saga'
+
+/**
+ * Fetch reserve parameters of the bonding curve.
+ * @returns {object} { totalETH, totalPNK, spread } all keys map to big number objects.
+ */
+function* fetchBondingCurveTotals() {
+  return {
+    totalETH: yield call(bondingCurve.getTotalETH),
+    totalPNK: yield call(bondingCurve.getTotalTKN),
```

Comment:
This could be optimized into one call.

Reply:
By making the requests in parallel or making a view function that returns all 3.